lib/src/Uncertainty/Distribution/MarginalUniformOrderStatistics.cxx

```diff
@@ -0,0 +1,194 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The MarginalUniformOrderStatistics distribution
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/MarginalUniformOrderStatistics.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/SimplicialCubature.hxx"
+#include "openturns/CubaIntegration.hxx"
+#include "openturns/GaussLegendre.hxx"
+#include "openturns/IntervalMesher.hxx"
+#include "openturns/UniformOrderStatistics.hxx"
+#include "openturns/DistFunc.hxx"
+#include "openturns/PlatformInfo.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(MarginalUniformOrderStatistics)
+
+static const Factory<MarginalUniformOrderStatistics> Factory_MarginalUniformOrderStatistics;
+
+
+/* Default constructor */
+MarginalUniformOrderStatistics::MarginalUniformOrderStatistics()
+  : MarginalDistribution(UniformOrderStatistics(1), {0})
+{
+  setName("MarginalUniformOrderStatistics");
+}
+
+/* Parameters constructor */
+MarginalUniformOrderStatistics::MarginalUniformOrderStatistics(const UnsignedInteger n,
+                                                               const Indices & indices)
+  : MarginalDistribution(UniformOrderStatistics(n), indices)
+{
+  setName("MarginalUniformOrderStatistics");
+  // compute the log-PDF normalization factor
+  // Waiting for a more accurate computation...
+  const UnsignedInteger m = indices.getSize();
+  if (m == 0) throw InvalidArgumentException(HERE) << "Error: cannot build a MarginalUniformOrderStatistics based on an empty Indice";
+  if (!indices.isStrictlyIncreasing()) throw InvalidArgumentException(HERE) << "Error: expected an increasing list of marginal indices, here indices=" << indices;
```

Comment:
We may use `indices.check(n)` to check that indices are in [0, ..., n - 1].

Reply:
True, good catch.

Reply:
done